File: ticketProject7-CrossPlatform/TS-cross/TicketModel/src/main/java/ticket/model/validators/BuyerValidator.java
package ticket.model.validators;

import ticket.model.Buyer;

import java.security.InvalidParameterException;


/**
 * Validates an instance of class Buyer
 * Throws ValidateException if instance is invalid
 */
public class BuyerValidator implements Validator<Buyer> {

    /**
     * @throws InvalidParameterException if id  id is null or <=0 / first name/last name are null.
     */
    @Override
    public void validate(Buyer entity) throws ValidationException {
        if (entity == null)
            throw new InvalidParameterException();
        else {
            if (entity.getId() <= 0)
                throw new ValidationException("Invalid ID!");
            if (entity.getIdAR() == null)
                throw new ValidationException("Invalid idAR!");
            if (entity.getFirstName() == null)
                throw new ValidationException("Invalid first name!");
            if (entity.getLastName() == null)
                throw new ValidationException("Invalid last name!");
            if (entity.getNoTickets() < 0)
                throw new ValidationException("Invalid noTichets!");
        }
    }
}

File: ticketProject5-DistributedC/TicketSystemCC/model/model/Validators/BuyerValidator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace model.Validators
{
    /**
     * Validates an instance of class Buyer
     * Throws ValidateException if instance is invalid
     */
    public class BuyerValidator : Validator<Buyer>
    {
        virtual public void Validate(Buyer entity)
        {
            if (entity == null)
                throw new ArgumentNullException("entity must be not null");
            else
            {
                if (entity.Id <= 0)
                    throw new ValidationException("Invalid ID!");
                if (entity.IdAR <= 0)
                    throw new ValidationException("Invalid IDAR!");
                if (entity.FirstName == null)
                    throw new ValidationException("Invalid first name!");
                if (entity.LastName == null)
                    throw new ValidationException("Invalid last name!");
                if (entity.NoTickets == 0)
                    throw new ValidationException("Invalid username!");
            }
        }
    }
}
